Amazon/Marketplace/v7153/templates/jfrog-artifactory-core-infrastructure.template.yaml

```diff
@@ -0,0 +1,408 @@
+      SubnetIds: !Ref SubnetIds
+  ArtifactoryDatabase:
+    Type: AWS::RDS::DBInstance
+    Properties:
+      AllocatedStorage: !Ref DatabaseAllocatedStorage
+      MultiAZ: !Ref MultiAzDatabase
+      Engine: !Ref DatabaseEngine
+      EngineVersion: !FindInMap
+        - DatabaseMap
+        - !Ref DatabaseEngine
+        - DatabaseVersion
+      MasterUsername: !Ref DatabaseUser
+      MasterUserPassword: !Ref DatabasePassword
+      DBInstanceClass: !Ref DatabaseInstance
+      DBName: !Ref DatabaseName
+      DBSubnetGroupName: !Ref  ArtifactoryDatabaseSubnetGroup
+      StorageEncrypted: true
+      VPCSecurityGroups:
+        - !Ref ArtifactoryDatabaseSG
+  ArtifactoryDatabaseSG:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      Tags:
+        - Key: Name
+          Value: artifactory-rds-sg
+      GroupDescription: SG for RDS Instance to allow communication from the Bastion and Artifactory servers.
+      VpcId: !Ref VpcId
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CidrIp: !Ref VpcCidr
+        - IpProtocol: tcp
+          FromPort: !FindInMap
+            - DatabaseMap
+            - !Ref DatabaseEngine
+            - port
+          ToPort: !FindInMap
+            - DatabaseMap
+            - !Ref DatabaseEngine
+            - port
+          CidrIp: !Ref PrivateSubnet1Cidr
+        - IpProtocol: tcp
+          FromPort: !FindInMap
+            - DatabaseMap
+            - !Ref DatabaseEngine
+            - port
+          ToPort: !FindInMap
+            - DatabaseMap
+            - !Ref DatabaseEngine
+            - port
+          CidrIp: !Ref PrivateSubnet2Cidr
+        - IpProtocol: tcp
+          FromPort: !FindInMap
+            - DatabaseMap
+            - !Ref DatabaseEngine
+            - port
+          ToPort: !FindInMap
+            - DatabaseMap
+            - !Ref DatabaseEngine
+            - port
+          CidrIp: !Ref PrivateSubnet3Cidr
+      SecurityGroupEgress:
+        - IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CidrIp: 0.0.0.0/0
+        - IpProtocol: tcp
+          FromPort: 80
+          ToPort: 80
+          CidrIp: 0.0.0.0/0
+        - IpProtocol: tcp
+          FromPort: 443
+          ToPort: 443
+          CidrIp: 0.0.0.0/0
+  ArtifactoryS3Bucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      AccessControl: Private
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: AES256
+  ArtifactoryS3IAMPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: S3BucketPermissions
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Sid: S3BucketPermissions
+            Effect: Allow
+            Action:
+              - s3:*
+            Resource:
+              - Fn::Join:
+                - ''
+                - - !Sub "arn:${AWS::Partition}:s3:::"
```

Comment:
You can simplify this to : `!Sub "arn:${AWS::Partition}s3:::${ArtifactoryS3Bucket}"`

Which reminds me: you definitely need to add some entropy to the name of this S3 bucket because their names are global